vumi/transports/dmark/tests/test_dmark_ussd.py

```diff
@@ -102,6 +105,43 @@ def test_inbound_begin(self):
         self.assert_ack(ack, reply)
 
     @inlineCallbacks
+    def test_inbound_status(self):
+        d = self.tx_helper.mk_request()
+        [msg] = yield self.tx_helper.wait_for_dispatched_inbound(1)
+        self.tx_helper.dispatch_outbound(msg.reply('foo'))
+        yield d
+
+        [status] = yield self.tx_helper.get_dispatched_statuses()
+        self.assertEqual(status['status'], 'ok')
+        self.assertEqual(status['component'], 'request')
+        self.assertEqual(status['type'], 'request_parsed')
+        self.assertEqual(status['message'], 'Request parsed')
+
+    @inlineCallbacks
+    def test_inbound_cannot_decode(self):
+        '''If the content cannot be decoded, an error shoould be sent back'''
+        user_content = "Who are you?".encode('utf-32')
+        response = yield self.tx_helper.mk_request(ussdRequestString=user_content)
+        self.assertEqual(response.code, 400)
+
+        body = json.loads(response.delivered_body)
+        request = body['invalid_request']
+        self.assertEqual(request['method'], 'GET')
```

Comment:
So there's a small issue in why I didn't do that. It seems that the URI is partially decoded (`%20` turned into `+`). I have no idea where this happens, or how to stop it, but it does mean that I cannot decode the URL and check it.